src/mantine-modals/src/ModalsProvider.tsx

```diff
@@ -61,8 +61,11 @@ export function ModalsProvider({ children, modalProps, labels, modals }: ModalsP
     props: null,
     type: 'content',
   });
-  const closeAll = () => {
-    state.forEach((modal) => modal.props?.onClose?.());
+  const closeAll = (canceled: boolean) => {
+    state.forEach((modal) => {
+      canceled && modal.props?.onCancel?.();
```

Comment:
TS is complaining that this prop does not exist in the modal props, which is part of the core modal component.